qubes/events.py

```diff
@@ -157,50 +160,92 @@ def _fire_event_in_order(self, order, event, kwargs):
                     key=(lambda handler: hasattr(handler, 'ha_bound')),
                     reverse=True):
                 effect = func(self, event, **kwargs)
+                # "is True" to rule out unittest.mock.Mock...
+                if getattr(func, 'ha_async', False) is True:
```

Comment:
That would also solve the problem with `emitter.add_handler()`. 👍 for removing those hacky coroutines without `@asyncio.coroutine`.